src/target/linux.rs

```diff
@@ -1,12 +1,39 @@
+use crate::target::thread::Thread;
+use crate::target::unix::{self, UnixTarget};
 use nix::unistd::{getpid, Pid};
+use procfs::process::Process;
 use std::{
     fs::File,
     io::{BufRead, BufReader},
     marker::PhantomData,
     mem,
 };
 
-use crate::target::unix::{self, UnixTarget};
+struct LinuxThread {
+    name: String,
+    id: i32,
+}
+
+impl LinuxThread {
+    fn new(name: &str, id: i32) -> LinuxThread {
```

Comment:
A minor suggestion: we can take `Into<String>` here to make the API more flexible:
```suggestion
    fn new(name: impl Into<String>, id: i32) -> LinuxThread {
        LinuxThread {
            name: name.into(),
            id,
        }
    }
```